hack/generator/pkg/astmodel/struct_type.go

```diff
@@ -103,6 +112,8 @@ func (structType *StructType) Equals(t Type) bool {
 			}
 		}
 
+		// TODO: Match funcs?
```

Comment:
Yeah, probably should for correctness - we might not need it now, but leaving it out is creating a landmine that might go off later.